### ISSUE TITLE
Calling the storage client many times returns concurrent.futures._base.TimeoutError

### ISSUE THREAD

**Issue by nabobalis** [talkiq/gcloud-aio#143]
I am unsure if this due to my inexperience with asyncio style code but in my current code, I end up calling `Storage` several times in quick succession then fetching metadata/blobs after and sometimes I get this error:

```
Backing off acquire_access_token(...) for 0.5s (concurrent.futures._base.TimeoutError)
```
which refers to: https://github.com/talkiq/gcloud-aio/blob/d090df40ed6e4cb759329cd5d4fdb486f8f73bac/auth/gcloud/aio/auth/token.py#L204 

In another case, I was trying to async gather a function that ended up calling a client per item in a list:
```
await asyncio.gather(*[check_file(bucket, file) for file in filelist])
```
Basically where `check_file` was a function given a bucket and a blob_name would check its hash in the metadata and return.

I was getting this same error. 

Unfortunately I don't have a simple code example right now but I can try to come up with one if it helps?

Locally I worked around it by creating fewer clients and passing them around which avoids this error.

**Comment by nabobalis** [comment]
I was using my own aiohttp.session, but if I let the client handle it instead, it seems to be holding.

**Comment by ZMarouani** [comment]
I'm encountering the same problem , would anyone please explain the exact cause of problem and the solution ?

**Comment by TheKevJames** [comment]
Basically, what's going on here is that the `gcloud-aio-*` libs will handle the aiohttp session management on their own if you _do not_ provide your own session manually (eg. via the constructor), but allow you to specify it if you'd like for more complicated use-cases (such as sharing a session across multiple clients).

To avoid any sort of oddities with session management, you basically want to stick to two rules:
* if the gcloud libraries created the session, the gcloud libraries should handle everything about the session (so, eg. don't close the session manually)
* if you created the session, you need to handle everything about it (so, eg. close the session when you're done)

Otherwise, I'd recommend letting `gcloud-aio` handle it all, but either method is 100% ok here.
